### PR TITLE
Next step suggestions

### PR DESCRIPTION
**Pull request by Cryaaa**
This is the implementation of the next step suggestions. As we discussed I added a parameter to the categories which includes a list of suggetsed category names for the next steps. I have added the next step suggestions purely based on my thoughts when looking at the categories and I think we should have a discussion about them before doing the final implementation (that's why this is a draft).

The highlighting of the next steps meant that a few changes to the button grid had to be made but the final result does not differ from the design much in my opinion. CUrrently highlighting is done by changing the background of the items to green which I think looks nice. During the development I noticed a bug which is fixed in #22 and thus should be merged beforehand. Finally I slightly modified the `get_category_of_function` function to allow only searching by function name, which makes life a bit easier for determining categories from loaded widgets.

### CODE REVIEW

**Comment by haesleinhuepf** [comment]
## [Codecov](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase) Report
Attention: Patch coverage is `64.81481%` with `19 lines` in your changes missing coverage. Please review.
> Project coverage is 30.18%. Comparing base [(`4612a38`)](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/commit/4612a38516a30eb4458251246a29552db9dab748?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase) to head [(`d6d6a23`)](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/commit/d6d6a23f24a85b15c8a5f7c2d0376f54b1c8c3a9?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase).
> Report is 107 commits behind head on main.

| [Files with missing lines](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase) | Patch % | Lines |
|---|---|---|
| [napari\_assistant/\_gui/\_Assistant.py](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&filepath=napari_assistant%2F_gui%2F_Assistant.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase#diff-bmFwYXJpX2Fzc2lzdGFudC9fZ3VpL19Bc3Npc3RhbnQucHk=) | 65.51% | [10 Missing :warning: ](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase) |
| [napari\_assistant/\_gui/\_button\_grid.py](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&filepath=napari_assistant%2F_gui%2F_button_grid.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase#diff-bmFwYXJpX2Fzc2lzdGFudC9fZ3VpL19idXR0b25fZ3JpZC5weQ==) | 72.22% | [5 Missing :warning: ](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase) |
| [napari\_assistant/\_categories.py](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&filepath=napari_assistant%2F_categories.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase#diff-bmFwYXJpX2Fzc2lzdGFudC9fY2F0ZWdvcmllcy5weQ==) | 40.00% | [3 Missing :warning: ](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase) |
| [napari\_assistant/\_workflow\_io\_utility.py](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&filepath=napari_assistant%2F_workflow_io_utility.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase#diff-bmFwYXJpX2Fzc2lzdGFudC9fd29ya2Zsb3dfaW9fdXRpbGl0eS5weQ==) | 0.00% | [1 Missing :warning: ](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase) |

<details><summary>Additional details and impacted files</summary>


```diff
@@            Coverage Diff             @@
##             main      #23      +/-   ##
==========================================
+ Coverage   28.89%   30.18%   +1.28%     
==========================================
  Files          13       13              
  Lines        1007     1047      +40     
==========================================
+ Hits          291      316      +25     
- Misses        716      731      +15     
```

</details>

[:umbrella: View full report in Codecov by Sentry](https://app.codecov.io/gh/haesleinhuepf/napari-assistant/pull/23?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase).   
:loudspeaker: Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Robert+Haase).

<details><summary> :rocket: New features to boost your workflow: </summary>

- :snowflake: [Test Analytics](https://docs.codecov.com/docs/test-analytics): Detect flaky tests, report on failures, and find test suite problems.
</details>

File: napari_assistant/_gui/_button_grid.py
Comment (haesleinhuepf) [inline napari_assistant/_gui/_button_grid.py:17]: I _think_ we may want to keep this...

File: napari_assistant/_gui/_button_grid.py
Comment (haesleinhuepf) [inline napari_assistant/_gui/_button_grid.py:60]: I _think_ we may want to remoe this to restore the old layout. But I'm speculating...

File: napari_assistant/_gui/_category_widget.py
Comment (haesleinhuepf) [inline napari_assistant/_gui/_category_widget.py:36]: Great, thanks for this! Very good style

File: napari_assistant/_gui/_button_grid.py
Comment (haesleinhuepf) [inline napari_assistant/_gui/_button_grid.py:73]:
```suggestion
        item.setBackground(_get_background_brush())
```

File: napari_assistant/_categories.py
Comment (haesleinhuepf) [inline napari_assistant/_categories.py:565]: That's smart! And I like that it's backwards compatible :-)

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hi Ryan @Cryaaa ,

this is so amazing! Thanks for working on this! Just for the record, the layout looks a bit differently (new version left, old right):

image

I think it's just button sizes and/or margins between the buttons. Could you take a look into this? 

Also, I think the highlight should not be so agressive green, but better a slightly more bright grey. The user should perceive it as a soft hint.

You see that's just minor suggestions. Otherwise I love it!! 

I can also update the suggested categories after merging, as it's tricky to send a PR to this branch. Next time, Feel free to create a branch on my repository, you should have write access.

Thanks again! This is so awesome!

Best,
Robert

File: napari_assistant/_gui/_button_grid.py
Comment (Cryaaa) [reply napari_assistant/_gui/_button_grid.py:17]: So I tried to keep as many things from the style sheet as possible but this line was somehow blocking that I could set the background color of the buttons. I guess we might need to find an alternative solution but I need to build up my courage to dive deeper into Qt stuff again :D

File: napari_assistant/_gui/_category_widget.py
Comment (Cryaaa) [reply napari_assistant/_gui/_category_widget.py:36]: No problem! I'm trying to do this as much as possible since I've shot myself in the foot (so to speakt) too many times now :)

File: napari_assistant/_gui/_button_grid.py
Comment (Cryaaa) [reply napari_assistant/_gui/_button_grid.py]: In the latest commit I changed the value slightly to match the old layout. You can see a screenshot in the latest comment!

**Comment by haesleinhuepf** [comment]
Awesome, I go for the darker option and see how it looks in the wild. Thanks again!

**Pull request merged**
